tools/target.py

```diff
@@ -53,19 +51,31 @@ def run(target, build, should_pull, **kwargs):
             != 0
         )
 
-        # If there are changes in tracked files in the docker folder when comparing with master,
-        # we want to build the image and we don't want to pull the image
-        if changed:
-            build = True
-            should_pull = False
-
-        if should_pull:
+        if pull:
             # Pull the image from dockerhub
             platform.pull(defaults.image, target)
 
-        # If user wants or needs to build it locally, do that
+        # If user wants or needs to build it locally, do that unless the build context is the same as origin/master
         if build:
-            platform.build(defaults.image, target)
+            if not changed:
+                cprint(
+                    "There aren't any changes to be made to the docker image. "
+                    "Skipping build, and using tagged image instead",
+                    "red",
+                    attrs=["bold"],
+                )
+            else:
+                platform.build(defaults.image, target)
+        # If the user hasn't selected build but they have changed the docker directory, then we warn them that the image
+        # isn't going to be built, even though the dirty diff implies they might want it to be built
```

Comment:
What is a diff? and why is a dirty one bad?

Reply:
Dirty diff here is meaning that your local `docker` directory is different to the `docker` directory that is on `origin/master`. In this case you would usually want to rebuild a docker image with your local changes, if you decide not too, a warning is displayed.